internal/router/router.go

```diff
@@ -896,9 +873,28 @@ func splitId(path string) (url *charm.URL, rest string, err error) {
 	// and path[0:i] should contain the entire
 	// charm id.
 	urlStr := strings.TrimSuffix(path[0:i], "/")
-	url, err = charm.ParseURL(urlStr)
+	url, err = parseURL(urlStr)
 	if err != nil {
 		return nil, "", errgo.Mask(err)
 	}
 	return url, path[i:], nil
 }
+
+func mustParseURL(s string) *charm.URL {
```

Comment:
Surely we don't need this. It would be better to take channels out of charm, until we do that why not just ignore any channel in a URL?